region/norfair/west/Ice Beam Gate Room.json

```diff
@@ -413,21 +327,107 @@
       "link": [3, 3],
       "name": "Crystal Flash",
       "requires": [
-        {"enemyKill": {
-          "enemies": [["Sm. Dessgeega"]],
-          "explicitWeapons": [
-            "Missile",
-            "Super",
-            "PowerBomb",
-            "ScrewAttack",
-            "Wave",
-            "Spazer",
-            "Plasma"
-          ]
-        }},
+        {"or": [
+          {"obstaclesCleared": ["D"]},
+          {"enemyDamage": {
+            "enemy": "Sm. Dessgeega",
+            "type": "contact",
+            "hits": 1
+          }},
+          {"enemyKill": {
+            "enemies": [["Sm. Dessgeega"]],
+            "explicitWeapons": [
+              "Missile",
+              "Super"
+            ]
+          }}
+        ]},
         "h_canCrystalFlash"
       ],
-      "clearsObstacles": ["C"]
+      "note": "Kill some or all of the enemies before Crystal Flashing."
+    },
+    {
+      "link": [3, 7],
+      "name": "Enemies Already Killed",
+      "requires": [
+        {"obstaclesCleared": ["D"]}
+      ]
+    },
+    {
+      "link": [3, 7],
+      "name": "Enemy Kill",
+      "requires": [
+        {"or": [
+          "Spazer",
+          "Plasma",
+          "ScrewAttack",
+          "Wave",
+          {"and": [
+            "canDodgeWhileShooting",
+            "h_canUsePowerBombs"
+          ]},
+          {"enemyKill": {
+            "enemies": [["Sm. Dessgeega"], ["Sm. Dessgeega"]],
+            "explicitWeapons": [
+              "Missile",
+              "Super"
+            ]
+          }},
+          {"and": [
+            "canDodgeWhileShooting",
+            {"enemyDamage": {
+              "enemy": "Sm. Dessgeega",
+              "type": "contact",
+              "hits": 1
+            }}
+          ]},
+          {"enemyDamage": {
+            "enemy": "Sm. Dessgeega",
+            "type": "contact",
+            "hits": 2
+          }}
+        ]}
+      ],
+      "clearsObstacles": ["D"]
+    },
+    {
+      "link": [3, 7],
+      "name": "Careful Kill",
+      "requires": [
+        "canDodgeWhileShooting",
+        "canTrickyJump"
+      ],
+      "clearsObstacles": ["D"],
+      "note": "Avoiding damage from all enemies with just Power Beam is tricky, but doable."
```

Comment:
`StopOnADime` or walk back to the door for more space to kill the first desgeega with.